rotkehlchen/db/history_events.py

```diff
@@ -397,39 +401,55 @@ def get_history_events(
+        if ignore_asset_filter != '':
+            ignore_asset_filter = f' WHERE event_identifier NOT IN (SELECT event_identifier FROM history_events {ignore_asset_filter})'  # noqa: E501
 
-        if has_premium is True:
-            if group_by_event_ids is True:
-                base_prefix += 'COUNT(*), '
-            base_query = f'{base_prefix} {base_suffix}'
+        if group_by_event_ids:
+            type_idx = 1
+            filters, filters_bindings = filter_query.prepare(
+                with_order=True,
+                with_group_by=True,
+                with_pagination=False,
+                without_ignored_asset_filter=True,
+            )
         else:
-            if group_by_event_ids is True:
-                special_free_query = True  # a bit ugly conditions to keep limit at groups for free users  # noqa: E501
-                free_query_group_by += 'GROUP BY event_identifier'
-                base_query = (
-                    f'{base_prefix} * FROM (SELECT COUNT(*), {base_suffix} '
-                    f'{free_query_group_by} ORDER BY timestamp DESC, sequence_index ASC LIMIT ?) '
+            filters, filters_bindings = filter_query.prepare(with_pagination=False)
+
+        if has_premium:
+            if group_by_event_ids:
+                base_query = group_by_event_id_query.format(
+                    exclude_ignore_assets=ignore_asset_filter,
+                    filters=filters,
+                    limit='',
                 )
-            else:  # if we don't group them but still apply free limit, then apply this limit only on any event_identifiers that fall inside the limit  # noqa: E501
-                base_query = (
-                    f'{base_prefix} * FROM (SELECT {base_suffix} WHERE event_identifier IN '
-                    f'(SELECT DISTINCT event_identifier FROM history_events {free_query_group_by} ORDER BY timestamp DESC, sequence_index ASC LIMIT ?)) '  # noqa: E501
+            else:
+                base_query = f'{base_prefix} {base_suffix} {filters}'
+
+            bindings.extend(filters_bindings)
+        else:
+            if group_by_event_ids:
+                base_query = group_by_event_id_query.format(
+                    exclude_ignore_assets=ignore_asset_filter,
+                    filters=filters,
+                    limit='LIMIT ?',
                 )
-            bindings = [FREE_HISTORY_EVENTS_LIMIT]
+            else:  # if we don't group them but still apply free limit, then apply this limit only on any event_identifiers that fall inside the limit  # noqa: E501
+                base_query = f'{base_prefix} * FROM (SELECT {base_suffix} {filters} LIMIT ?)'
 
-        prepared_query, prepared_bindings = filter_query.prepare(
-            with_group_by=group_by_event_ids,
-            special_free_query=special_free_query,
-        )
-        bindings.extend(prepared_bindings)
+            bindings.extend(filters_bindings + [FREE_HISTORY_EVENTS_LIMIT])
```

Comment:
you can do this extend at line 425 and here just do it 

```suggestion
            bindings.extend([FREE_HISTORY_EVENTS_LIMIT])
```